src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanContext.hpp

```diff
@@ -208,6 +211,57 @@ struct TScanContext final
             // Clear the cpeName if the OS is not supported
             m_osData.cpeName = "";
         }
+
+        // Build CPE for generic cases or running OS when build the CPE name.
+        buildCPENameForGeneric();
+    }
+
+    /**
+     * @brief Build CPE for generic cases or running OS.
+     */
+    void buildCPENameForGeneric()
+    {
+        // Get references to global data maps
+        const auto& osCpeMaps = TGlobalData::instance().osCpeMaps();
+        const auto& reverseVendorMaps = TGlobalData::instance().vendorMaps();
+
+        // Initialize the CPE name
+        m_osData.cpeGeneric = "cpe:/o:";
+
+        if (!(osCpeMaps.contains(KEY_GENERIC_CPE) && reverseVendorMaps.contains(KEY_REVERSE_VENDOR)))
+        {
+            // Clear the CPE name if the OS is not supported
+            m_osData.cpeGeneric.clear();
+            return;
+        }
+
+        // Get the generic CPE from the OS CPE maps
+        std::string cpeTemplate = osCpeMaps.at(KEY_GENERIC_CPE);
+        std::string vendorName;
+
+        // Find the vendor name in the reverse vendor maps
+        for (const auto& item : reverseVendorMaps[KEY_REVERSE_VENDOR])
```

Comment:
Its preferable to produce an exception than a segfault:
```suggestion
        for (const auto& item : reverseVendorMaps.at(KEY_REVERSE_VENDOR))
```